ner/scripts/paramsweep/generatesweep.sh

```diff
@@ -5,15 +5,151 @@
 # alternative learning rates and thicknesses. Note that the defaults
 # are overriden by the configuration arguments for both models.
 #
-learnrate=(.2 .01)
-thickness=(20 200)
 
-for i in ${learnrate[@]}; do
-    for j in ${thickness[@]}; do
-	    NEWDIR="r"$i"-t"$j
-	    cp -rf ./cogcomp-nlp/ ./$NEWDIR
-	    NEWNERDIR=$NEWDIR"/ner"
-	    sed 's/SparseAveragedPerceptron\([^\)]*\)/SparseAveragedPerceptron\('"$i"', 0, '"$j"'/' $NEWNERDIR/src/main/lbj/LbjTagger.lbj > $NEWNERDIR/src/main/lbj/poogers.lbj
-	    mv $NEWNERDIR/src/main/lbj/poogers.lbj $NEWNERDIR/src/main/lbj/LbjTagger.lbj
-    done;
-done;
\ No newline at end of file
+#
+# first build cogcomp-nlp. Oddly enought, take much long to build than to
+# move it already built.
+#
+echo "Building cogcomp-nlp"
+bash -c "cd cogcomp-nlp ; mvn package -DskipTests dependency:copy-dependencies"
+
+#
+# we don't need the contents of any of the build directories but those in the ner directory.
+bash -c "cd cogcomp-nlp/temporal-normalizer ; mvn clean"
+bash -c "cd cogcomp-nlp/verbsense ; mvn clean"
+bash -c "cd cogcomp-nlp/big-data-utils ; mvn clean"
+bash -c "cd cogcomp-nlp/core-utilities ; mvn clean"
+bash -c "cd cogcomp-nlp/pipeline ; mvn clean"
+bash -c "cd cogcomp-nlp/prepsrl ; mvn clean"
+bash -c "cd cogcomp-nlp/corpusreaders ; mvn clean"
+bash -c "cd cogcomp-nlp/quantifier ; mvn clean"
+bash -c "cd cogcomp-nlp/corpusreaders ; mvn clean"
+bash -c "cd cogcomp-nlp/curator ; mvn clean"
+bash -c "cd cogcomp-nlp/commasrl ; mvn clean"
+bash -c "cd cogcomp-nlp/external/stanford_3.8.0 ; mvn clean"
+bash -c "cd cogcomp-nlp/external/external-commons ; mvn clean"
+bash -c "cd cogcomp-nlp/external/path-lstm ; mvn clean"
+bash -c "cd cogcomp-nlp/external/clausie ; mvn clean"
+bash -c "cd cogcomp-nlp/external/stanford_3.3.1 ; mvn clean"
+bash -c "cd cogcomp-nlp/tokenizer ; mvn clean"
+bash -c "cd cogcomp-nlp/similarity ; mvn clean"
+bash -c "cd cogcomp-nlp/lbjava-nlp-tools ; mvn clean"
+bash -c "cd cogcomp-nlp/inference ; mvn clean"
+bash -c "cd cogcomp-nlp/lemmatizer ; mvn clean"
+bash -c "cd cogcomp-nlp/depparse ; mvn clean"
+bash -c "cd cogcomp-nlp/edison ; mvn clean"
+bash -c "cd cogcomp-nlp/md ; mvn clean"
+bash -c "cd cogcomp-nlp/pos ; mvn clean"
+
+# learning rates and thickness range
+learnrateL1=(.01 .03)
+thicknessL1=(10 30)
+learnrateL2=(.02 .04)
+thicknessL2=(20 40)
+
+#
+# copy the code all over the place.
+#
+echo
+echo -------- cloning code into new directories --------------
```

Comment:
seems like this will take up a lot of space...  we may want to think about how to refine it.   Is it plausible to simply change output paths, or at least just create symlinks -- I'm assuming the run script launches separate processes for each parameter set; even in parallel, would this create any problems?

Reply:
I already know how to refine this, but way out of scope for this release. This is already *much* smaller that it was in the previous iteration. The right way to do this is to run them all in one VM. Makes the hairs on the back of my neck rise. Much smaller memory footprint, much faster, lots less disk.